### PR TITLE
Document Compose's syntax for variable defaults

### PR DESCRIPTION
**Pull request by nVitius**
This feature was not supported previously, but
was added in https://github.com/docker/compose/pull/3108

### CODE REVIEW

**Comment by aanand** [comment]
Thanks! We should document both forms of the inline default syntax:

- `${VAR-default}` (only fall back if VAR is _unset_)
- `${VAR:-default}` (fall back if VAR is _unset or empty_)

**Comment by johndmulhausen** [comment]
Question: Is this for upcoming Docker Compose functionality or is this functionality already released?

**Comment by nVitius** [comment]
@aanand: thanks, I missed that part.
@johndmulhausen: It looks like it's slated for 1.9.

**Comment by mstanleyjones** [comment]
In that case, please rebase this on the branch `vnext-compose`.

**Comment by nVitius** [comment]
Okay, I rebased over the `vnext-compose` branch and added the comment regarding the different syntax.

File: compose/compose-file.md [hunk 1]
```diff
@@ -1143,8 +1143,14 @@ string. In the example above, if `EXTERNAL_PORT` is not set, the value for the
 port mapping is `:5000` (which is of course an invalid port mapping, and will
 result in an error when attempting to create the container).
 
-Both `$VARIABLE` and `${VARIABLE}` syntax are supported. Extended shell-style
-features, such as `${VARIABLE-default}` and `${VARIABLE/foo/bar}`, are not
+In the case of environment variables that are not set, it is also possible to
```
Comment (mstanleyjones) [inline compose/compose-file.md]: Consider "you can" instead of "it is possible to".

File: compose/compose-file.md [hunk 2]
```diff
@@ -1143,8 +1143,14 @@ string. In the example above, if `EXTERNAL_PORT` is not set, the value for the
...
+define default values like so: `${VARIABLE:-default}`. In this case, Compose
```
Comment (mstanleyjones) [inline compose/compose-file.md]: Consider rewriting this a bit:

...define default values using one of the following syntax variants.

- To provide a default value which will be used if `VARIABLE` is unset or empty:

   ```
   ${VARIABLE:-default}
   ```

- To provide a default value if `VARIABLE` is empty:
   ```bash
   ${VARIABLE-default}
   ```

Both...

**Comment by londoncalling** [comment]
@mstanleyjones can you look at the updates I made here per your comments, and if this looks good, review and approve? Then either you or I can merge this...?

**Comment by mstanleyjones** [comment]
LGTM, thought I would still reconsider "it is possible to"...

**Pull request merged**